### PR TITLE
Resolve certain warning messages on Visual Studio

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/cocos-engine/issues/14273

CMake warnings will be redirected to the editor in a `console.log` style to minimize distraction to the user. However, this approach also poses the risk of missing important information. Nevertheless, we can analyze the build logs to mitigate this potential issue.

### Changelog
- Lower the level of CMake warnings.
- Remove empty files and resolve `LNK4221` warning message, which reads: 
``` 
warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that consumes this library.
```

  *This doesn't address all the similar errors. Some files are empty and were generated automatically, and they cannot be removed in this change.*
- Simultaneously remove empty files from CMakeLists.txt.
- Fix a few warnings concerning integer type conversions.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/CMakeLists.txt [hunk 1]
```diff
@@ -2033,8 +1940,8 @@ cocos_source_files(
     cocos/core/utils/IDGenerator.h
     cocos/core/utils/Path.cpp
     cocos/core/utils/Path.h
-    cocos/core/utils/MutableForwardIterator.cpp
-    cocos/core/utils/MutableForwardIterator.h
+    # cocos/core/utils/MutableForwardIterator.cpp
+    # cocos/core/utils/MutableForwardIterator.h
```
Comment (minggo) [inline native/CMakeLists.txt]: Why comment it?

File: native/cocos/audio/common/utils/primitives.cpp [hunk 2]
```diff
@@ -317,7 +317,7 @@ size_t nonZeroStereo16(const int16_t *frames, size_t count) {
  */
 #define COPY_FRAME_BY_MASK(dst, dmask, src, smask, count, zero) \
     {                                                           \
-        uint32_t bit, ormask;                                   \
+        int32_t bit, ormask;                                    \
```
Comment (minggo) [inline native/cocos/audio/common/utils/primitives.cpp:320]: Why need to change to int32_t?

File: native/cocos/audio/common/utils/primitives.cpp [hunk 3]
```diff
@@ -437,8 +437,8 @@ size_t memcpy_by_index_array_initialization(int8_t *idxary, size_t idxcount,
                                             uint32_t dstMask, uint32_t srcMask) {
     size_t n = 0;
     int srcidx = 0;
-    uint32_t bit;
-    uint32_t ormask = srcMask | dstMask;
+    int32_t bit;
+    int32_t ormask = srcMask | dstMask;
```
Comment (minggo) [inline native/cocos/audio/common/utils/primitives.cpp:441]: The same as above.

File: native/cocos/audio/common/utils/primitives.cpp (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/audio/common/utils/primitives.cpp:320]: A warning is generated when taking the negative of an unsigned int variable, but in this case, both variables are used for bitwise operations and do not impact the result.

File: native/CMakeLists.txt (hunk 1 above)
Comment (PatriceJiang) [reply native/CMakeLists.txt]: These two files did not participate in the compilation.

Comment (minggo) [reply native/CMakeLists.txt]: If so, why not delete these files too?

**Review by minggo — APPROVED** [review]

**Pull request merged**